src/firefly/java/edu/caltech/ipac/util/FitsHDUUtil.java

```diff
@@ -29,6 +33,74 @@
  * @version $Id: VoTableUtil.java,v 1.4 2013/01/07 22:10:01 tatianag Exp $
  */
 public class FitsHDUUtil {
+    private static final List<String> NAXIS_SET = Arrays.asList("naxis", "naxis1", "naxis2", "naxis3");
+
+    public static FileAnalysis.Report analyze(File infile, FileAnalysis.ReportType type) throws Exception {
+        FileAnalysis.Report report = new FileAnalysis.Report(type, infile.length(), infile.getPath());
```

Comment:
I am glad you broke this out. I needs work but it is clear and easy what could be added.  Looking at this I think stil think the `Part` needs a optional forth parameter.